src/macros.rs

```diff
@@ -16,18 +16,15 @@ macro_rules! align_up {
 ///
 /// From http://blog.phil-opp.com/rust-os/printing-to-screen.html, but tweaked
 /// for HermitCore.
-#[macro_export]
 macro_rules! print {
-	($($arg:tt)+) => ({
-		$crate::_print(format_args!($($arg)*));
-	});
+	($($arg:tt)*) => ($crate::_print(format_args!($($arg)*)));
 }
 
 /// Print formatted text to our console, followed by a newline.
-#[macro_export]
 macro_rules! println {
-	() => ($crate::print!("\n"));
-	($($arg:tt)+) => ($crate::print!("{}\n", format_args!($($arg)+)));
+	() => (print!("\n"));
+	($s:expr) => (print!(concat!($s, "\n")));
```

Comment:
What is this specialization used for? This allows `println!(3)`, which we should discourage. `std`'s `println` requires the format argument to be a string literal.

```suggestion
```

Reply:
Fun fact... Without this line, my code doesn't work on aarch64. Hm, in general does the logging macro not work. I have to check it.